File: 0771-jewels-and-stones/0771-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var answer = 0
        val jewelsSet = jewels.toSet()

        for (stone in stones) {
            if (jewelsSet.contains(stone)) {
                answer += 1
            }
        }
        return answer
    }
}

File: 0771-jewels-and-stones/0771-jewels-and-stones.py
from collections import defaultdict, Counter

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        answer = 0
        jewels_dict = Counter(jewels)

        for stone in stones:
            if jewels_dict[stone]:
                answer += 1
        return answer